packages/wrangler/src/inspect.ts

```diff
@@ -51,25 +52,19 @@ interface InspectorProps {
 export default function useInspector(props: InspectorProps) {
   /** A unique ID for this session. */
   const inspectorIdRef = useRef(randomId());
+
+  /** The websocket from the devtools instance. */
+  const [localWebSocket, setLocalWebSocket] = useState<WebSocket>();
+  /**  The websocket from the edge */
+  const [remoteWebSocket, setRemoteWebSocket] = useState<WebSocket>();
+
   /**
    *  The local proxy server that acts as the bridge between
    *  the remote websocket and the local DevTools instance.
    */
-  const serverRef = useRef<Server>();
-  /** The websocket server that runs on top of the proxy server. */
-  const wsServerRef = useRef<WebSocketServer>();
-
-  /** The websocket from the devtools instance. */
-  const [localWebSocket, setLocalWebSocket] = useState<WebSocket | undefined>();
-  /**  The websocket from the edge */
-  const [remoteWebSocket, setRemoteWebSocket] = useState<
-    WebSocket | undefined
-  >();
-
-  if (!serverRef.current) {
-    // Let's create the proxy server!
-    serverRef.current = createServer(
-      (req: IncomingMessage, res: ServerResponse) => {
+  const server = useMemo<Server>(
```

Comment:
This is not a good idea, because useMemo is not guaranteed to hold the cache indefinitely. 

![image](https://user-images.githubusercontent.com/18808/149755086-69e58752-d86c-4c48-9884-8be20772d517.png)